### PR TITLE
Fixed the outColor attribute

### PR DESCRIPTION
**Pull request by ppenenko**
Maya expects surface nodes to have an outColor output attribute and complains if it's missing when creating our nodes in Hypershade.

It's also created statically now and no matter how the node is created (previously dynamically and from the command).

Also some cleanup.

### CODE REVIEW

File: source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp [hunk 1]
```diff
@@ -48,73 +41,53 @@ const MString MaterialXSurfaceNode::MATERIALX_SURFACE_NODE_TYPENAME("MaterialXSu
...
-        CHECK_MSTATUS(nAttr.setHidden(false));
-
-        mdgModifier.addAttribute(thisMObject(), _outAttr);
-    }
-}
+    MFnNumericAttribute nAttr;
+    MObject outColorR = nAttr.create(outColorName + "R", "ocr", MFnNumericData::kFloat, 0.0);
+    MObject outColorG = nAttr.create(outColorName + "G", "ocg", MFnNumericData::kFloat, 0.0);
+    MObject outColorB = nAttr.create(outColorName + "B", "ocb", MFnNumericData::kFloat, 0.0);
+    OUT_ATTRIBUTE = nAttr.create(outColorName, "oc", outColorR, outColorG, outColorB);
```
Comment (ppenenko) [inline source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp]: The new output attribute.

File: source/MaterialXGenOgsXml/OgsXmlGenerator.cpp [hunk 2]
```diff
@@ -250,7 +250,7 @@ namespace
     }
 }
 
-const string OgsXmlGenerator::OUTPUT_NAME = "out";
+const string OgsXmlGenerator::OUTPUT_NAME = "outColor";
```
Comment (bernardkwok) [inline source/MaterialXGenOgsXml/OgsXmlGenerator.cpp]: It's not really a color all of the time, but if this makes it work guess we can leave it as this for now.

**Comment by bernardkwok** [comment]
I'm wondering if this change is needed if you override the method:
MPxSurfaceShadingNodeOverride::primaryColorParameter() ?

Also can we set up the output to be a single 4 channel attribute and use:
MPxSurfaceShadingNodeOverride::transparencyParameter() ?
as we need trasparency support and this would be the easiest way to do so IMHO.

File: source/MaterialXGenOgsXml/OgsXmlGenerator.cpp (hunk 2 above)
Comment (ppenenko) [reply source/MaterialXGenOgsXml/OgsXmlGenerator.cpp]: Well, if it's a Maya surface shader DG node then it's expected to be a color. The type can definitely be arbitrary in other use cases, such when using the generated fragment with fragment graph API to construct an MShaderInstance directly.

**Comment by ppenenko** [comment]
Not sure about MPxSurfaceShadingNodeOverride::primaryColorParameter(), good question. On the other hand, outColor is consistent with all other surface DG node examples I could find, so I do believe it's a good solution for now.

**Pull request merged**